app/main.py

```diff
@@ -1 +1,29 @@
-# write your code here
+import json
+from decimal import Decimal
+
+
+def calculate_profit(name: str):
+    with open(name) as file_read:
+        trades_data = json.load(file_read)
+    money_dict = {}
+    coin_account = Decimal("0")
+    for trade in trades_data:
+        if trade["sold"] is None:
+            money_bought = Decimal(trade["bought"]) * \

```

Comment:
It is illogical to check if `trade["sold"]` is None and then operate with `trade["bought"]`.